debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/PostgresConnectorIT.java

```diff
@@ -695,6 +697,34 @@ public void shouldResumeSnapshotIfFailingMidstream() throws Exception {
         TestHelper.dropDefaultReplicationSlot();
     }
 
+    @Test
+    @FixFor("DBZ-1857")
+    // The column used to kill connection is only supported since Postgres version 10
+    @SkipWhenDatabaseVersionLessThan(POSTGRES_10)
+    public void shouldRecoverFromRetriableException() throws Exception {
+        // Testing.Print.enable();
+        String setupStmt = SETUP_TABLES_STMT;
+        TestHelper.execute(setupStmt);
+        Configuration config = TestHelper.defaultConfig()
+                .with(PostgresConnectorConfig.SNAPSHOT_MODE, SnapshotMode.INITIAL.getValue())
+                .with(PostgresConnectorConfig.DROP_SLOT_ON_STOP, Boolean.FALSE)
+                .build();
+
+        start(PostgresConnector.class, config);
+        assertConnectorIsRunning();
+        waitForStreamingRunning("postgres", TestHelper.TEST_SERVER);
+
+        assertRecordsFromSnapshot(2, 1, 1);
+
+        // kill all opened connections to the database
+        TestHelper.execute("SELECT pg_terminate_backend(pid) FROM pg_stat_activity WHERE backend_type='walsender'");
```

Comment:
I was using something slightly different:

```
SELECT  pg_terminate_backend(pid) FROM pg_stat_activity WHERE pid = (SELECT * FROM pg_stat_activity where datname = 'postgres');
```

Could work on PG 9, too, perhaps?

Reply:
@gunnarmorling Thats for th tip, I'll modified it a little bit but I hope it will now for for all versions

Reply:
@gunnarmorling I have to reintroduce it back. There is not a reliable way how to choose the correct PID in case of multiple activities, so let's keep tests only for postgres 10+

Reply:
Ok.